### PR TITLE
Added: the script for the pre-order app and gitignore file(shopify-preorder#01)

### PR DESCRIPTION
**Pull request by ymaheshwari1**
Issues: https://github.com/hotwax/shopify-preorder-app/issues/1

- Added the script for pre-order app
- Added a gitignore file
- Added package.json file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "shopify-preorder-app",
+  "version": "1.0.0",
+  "description": "HotWax Commerce shopify preorder app",
+  "main": "public/script-tag.js",
+  "scripts": {
```
Comment (adityasharma7) [inline package.json:6]: Add minification logic as well

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/hotwax/shopify-preorder-app.git"
+  },
+  "author": "HotWax Commerce",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply package.json]: Sir, updated the license.

File: public/script-tag.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+
+  (function () {
+    let jQueryBopis;
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply public/script-tag.js]: Sir, updated the variable name.

File: package.json (hunk 1 above)
Comment (ymaheshwari1) [reply package.json:6]: Sir, added the minification script.

File: public/script-tag.js [hunk 4]
```diff
@@ -81,14 +79,17 @@
...
 
             const ifPreOrderActive = checkPreOrder(sku);
 
-            if (ifPreOrderActive) {
+            if (ifPreOrderActive.preOrder) {
                 // will add Pre Order to the button
                 addToCartButton.html("Pre Order");
 
+                // will add a span tag to define the date from when the product starts the shipping
+                addToCartButton.after(`<span>Ships from ${ifPreOrderActive.timestamp}</span>`);
```
Comment (adityasharma7) [inline public/script-tag.js:91]: Hi @ymaheshwari1,
We will not add entire text after the button, instead find element with `hc_preordershipsfrom` id and add date to it in human readable format.

**Pull request merged**